erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -663,6 +672,9 @@ def validate_fixed_asset_account(self):
 
 	def on_recurring(self, reference_doc):
 		self.due_date = None
+	
+	def get_link_filters(self, for_doctype):
```

Comment:
Don't need to define this function in all individual files. Transaction Base class already inherited, so it is already available.